R/routing.R

```diff
@@ -484,7 +484,16 @@ Routing <- R6::R6Class(
           .globals$infoLog$log(req$REQUEST_METHOD, "on", req$PATH_INFO)
 
           # parse request
-          request$params <- set_params(request$PATH_INFO, private$.routes[[i]]$route)
+          request$params <- tryCatch(
+            set_params(request$PATH_INFO, private$.routes[[i]]$route),
+            error = function(error){
+              error
+            }
+          )
+
+          if(inherits(request$params, "error")){
+            return(private$.routes[[i]]$error(request, res, response))
```

Comment:
since we're dealing with the `request$params` object, this return value should be:

```r
private$.routes[[i]]$error(request, res, request$params)
```